src/components/Popover/__tests__/Popover.spec.tsx

```diff
@@ -0,0 +1,180 @@
+
+test.describe('Popover', () => {
+  test.describe('visual', () => {
+    [
+      ...propTests.defaultComponentPropTest,
+      ...placementPropTest,
+      ...popoverTargetIdPropTest,
+    ].forEach(({
+      name,
+      onBeforeTest,
+      onBeforeSnapshot,
+      props,
+    }) => {
+      test(name, async ({
+        mount,
+        page,
+      }) => {
+        if (onBeforeTest) {
+          await onBeforeTest(page);
+        }
+
+        const hasTargetId = props?.popoverTargetId !== undefined;
+
+        const component = hasTargetId
+          ? await mount(
+            <PopoverWithTargetIdForTest
+              {...props}
+            />,
+          )
+          : await mount(
+            <PopoverForTest
+              {...props}
+            />,
+          );
+
+        if (onBeforeSnapshot) {
+          await onBeforeSnapshot(page, component);
+        }
+
+        const screenshot = await component.screenshot();
+        expect(screenshot).toMatchSnapshot();
+      });
+    });
+  });
+
+  test.describe('non-visual', () => {
+    test('id', async ({ mount }) => {
+      const id = 'custom-id';
+      const childrenText = 'Some text';
+
+      const component = await mount(
+        <PopoverForTest id={id}>
+          {childrenText}
+        </PopoverForTest>,
+      );
+
+      await expect(component.getByText(childrenText)).toHaveAttribute('id', id);
+    });
+
+    test('ref', async ({ mount }) => {
+      const childrenText = 'Some text';
+
+      const component = await mount(
+        <PopoverForRefTest
+          testRefAttrName="test-ref"
+          testRefAttrValue="test-ref-value"
+        >
+          {childrenText}
+        </PopoverForRefTest>,
+      );
+
+      await expect(component.getByText(childrenText)).toHaveAttribute('test-ref', 'test-ref-value');
+    });
+  });
+
+  test.describe('functionality', () => {
+    test('render in portal when portalId defined', async ({
+      mount,
+      page,
+    }) => {
+      const portalId = 'portal-id';
+      const portalContent = 'portal content';
+
+      await page.evaluate(() => {
+        document.body.innerHTML += '<div id="portal-id" />';
+      });
+
+      await mount(
+        <PopoverForTest portalId={portalId}>
+          {portalContent}
+        </PopoverForTest>,
+      );
+
+      const portalHTMLContent = await page
+        .evaluate((id) => document.getElementById(id).innerHTML, portalId);
+
+      expect(portalHTMLContent).toContain(portalContent);
+    });
+
+    test('pass placementStyle into style property', async ({ mount }) => {
+      const content = 'content';
+      const insetPlacementStyle = {
+        inset: '10px',
+        'inset-block-end': 'auto',
+        'inset-block-start': 'auto',
+        'inset-inline-end': 'auto',
+        'inset-inline-start': 'auto',
+        position: 'absolute',
+        'transform-origin': 'center',
+        translate: '10px',
+      };
+
+      const positionStyleObject1 = {
```

Comment:
This is not visual test. Why to we need to have 3 different tests? Why `left`, `top`, `bottom`, `left` is not tested together inside `insetPlacementStyle`?